### ISSUE TITLE
Cannot render nested vector

### ISSUE THREAD

**Issue by petehayes102** [nickel-org/rust-mustache#41]
I'm trying to render a nested vector and I'm having no joy. The following code produces an error on v0.7 and an empty vector on HEAD:

``` rust
let template = mustache::compile_str("{{#.}}{{#.}}{{.}}{{/.}}{{/.}}");
let data = vec![vec!["val1", "val2"]];
let mut r = Vec::new();
template.render(&mut r, &data).unwrap();
assert_eq!(r, b"val1val2");
```

As far as I can tell, this is valid Mustache syntax, though admittedly I am new to the game. Btw the error on v0.7 is `panicked at 'expected map: [""]'`, if that is meaningful to someone?

Cheers!

**Comment by Ryman** [comment]
Sorry for the delay on this, I've opened #42 to take care of it.

**Comment by petehayes102** [comment]
Legend! Cheers mate.

**Comment by Ryman** [comment]
Published 0.8 which includes this.
